host/retry_policy_workflow_test.go

```diff
@@ -0,0 +1,319 @@
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package host
+
+import (
+	"bytes"
+	"strconv"
+	"time"
+
+	"github.com/pborman/uuid"
+	"github.com/uber/cadence/common"
+	"github.com/uber/cadence/common/log/tag"
+	"github.com/uber/cadence/common/types"
+)
+
+func (s *IntegrationSuite) TestWorkflowRetryPolicyTimeout() {
+	id := "integration-workflow-retry-policy-timeout-test"
+	wt := "integration-workflow-retry-policy-timeout-test-type"
+	tl := "integration-workflow-retry-policy-timeout-test-tasklist"
+	identity := "worker1"
+
+	workflowType := &types.WorkflowType{}
+	workflowType.Name = wt
+
+	taskList := &types.TaskList{}
+	taskList.Name = tl
+
+	retryPolicy := &types.RetryPolicy{
+		InitialIntervalInSeconds:    1,
+		BackoffCoefficient:          2.0,
+		MaximumIntervalInSeconds:    5,
+		ExpirationIntervalInSeconds: 5,
+		MaximumAttempts:             0,
+		NonRetriableErrorReasons:    []string{"bad-error"},
+	}
+	request := &types.StartWorkflowExecutionRequest{
+		RequestID:                           uuid.New(),
+		Domain:                              s.domainName,
+		WorkflowID:                          id,
+		WorkflowType:                        workflowType,
+		TaskList:                            taskList,
+		Input:                               nil,
+		ExecutionStartToCloseTimeoutSeconds: common.Int32Ptr(10),
+		TaskStartToCloseTimeoutSeconds:      common.Int32Ptr(10),
+		Identity:                            identity,
+		RetryPolicy:                         retryPolicy,
+	}
+	now := time.Now()
+	we, err0 := s.engine.StartWorkflowExecution(createContext(), request)
+	s.Nil(err0)
+
+	s.Logger.Info("StartWorkflowExecution", tag.WorkflowRunID(we.RunID))
+	time.Sleep(11 * time.Second) // wait 1 second for timeout
+
+	historyResponse, err := s.engine.GetWorkflowExecutionHistory(createContext(), &types.GetWorkflowExecutionHistoryRequest{
+		Domain: s.domainName,
+		Execution: &types.WorkflowExecution{
+			WorkflowID: id,
+		},
+	})
+	s.Nil(err)
+	history := historyResponse.History
+	s.NotNil(history.Events[0])
+	s.Equal(history.Events[0].EventType, types.EventTypeWorkflowExecutionStarted.Ptr())
+	expirationTime := history.Events[0].WorkflowExecutionStartedEventAttributes.ExpirationTimestamp
+	s.NotNil(expirationTime)
+	delta := time.Unix(0, *expirationTime).Sub(now)
+	s.True(delta > 5*time.Second)
+	s.True(delta < 6*time.Second)
+
+	wfExecution, err := s.engine.DescribeWorkflowExecution(createContext(), &types.DescribeWorkflowExecutionRequest{
+		Domain: s.domainName,
+		Execution: &types.WorkflowExecution{
+			WorkflowID: id,
+		},
+	})
+	s.NotNil(wfExecution)
+	delta = time.Unix(0, *wfExecution.WorkflowExecutionInfo.CloseTime).Sub(time.Unix(0, *wfExecution.WorkflowExecutionInfo.StartTime))
+	s.True(delta > 9*time.Second)
+	s.True(delta < 11*time.Second)
+}
+
+func (s *IntegrationSuite) TestWorkflowRetryPolicyContinueAsNewAsRetry() {
```

Comment:
nit: Do we still need the following two tests given that there's no logic change in the continue as new path?

Reply:
I don't think we have a test for these flows as far as i know, so thought i would add some